src/utilities/fe_mesh.jl

```diff
@@ -0,0 +1,186 @@
+# Typ definitions
+type FEBoundary{T}
+    topology::Array{Int}
+end
+
+type FEMesh{dim, T <: Real}
+    coords::Vector{Vec{dim,T}} # Node coordinates
+    topology::Matrix{Int} # Element nodes
+    boundary::Vector{Vector{}} # Boundary nodes
+end
+
+
+function FEMesh() # Input should be shape, vectors with start and end-coordinates, and a vector with number of elements in each dir.
+    # Set up the mesh here
+    # Do some input checks and stuff
+
+    # Here I was thinking we could call a set_up_mesh function that is overloaded for 1,2 and 3 dimensions somehow
+    # coords,topology,boundary = set_up_mesh(dim)
+
+    # return FEMesh(coords,topology,boundary)
+
+end
+
+
+#############
+# Accessors #
+#############
+
+# Should these even exist? Atleast I should add the same accessors for the FEDofs aswell so that contains all information needed
```

Comment:
But as it is now, the `FEMesh` does not know which fields it has. So getting the node coordinates to `reinit!` a function space which has quadratic approximation will be more awkward.